### PR TITLE
make UUID() a standard function

### PR DESCRIPTION
**Pull request by ericyuliu**
make UUID() a standard function

Cherry-pick of trinodb/trino#786

Co-authored-by: Piotr Findeisen <piotr.findeisen@gmail.com>
Co-authored-by: David Phillips <david@acz.org>

```
== RELEASE NOTES ==

General changes
* Add :func:`uuid` to return a unique identifier

Accumulo Connector Changes
* Replace :func:`uuid` from being connector specific to a standard function. In the process the return type has been changed from ``VARCHAR`` to ``UUID`` and existing use cases might need to perform a cast - ``CAST(UUID() AS VARCHAR)``.
```

### CODE REVIEW

File: presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloPlugin.java [hunk 1]
```diff
@@ -13,22 +13,18 @@
...
-    @Override
-    public Set<Class<?>> getFunctions()
-    {
-        return ImmutableSet.<Class<?>>builder().add(AccumuloStringFunctions.class).build();
-    }
+//    @Override
+//    public Set<Class<?>> getFunctions()
+//    {
+//        return ImmutableSet.<Class<?>>builder().add(AccumuloStringFunctions.class).build();
+//    }
```
Comment (mayankgarg1990) [inline presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloPlugin.java:27]: delete instead of commenting ?

**Review by mayankgarg1990 — CHANGES_REQUESTED** [review]
The commit and PR title should convey the intent and the fact that it is a cherry pick can go in the commit message.

Commit title and PR title - `Make UUID() a standard function`

In addition to that based on my understanding - for cherry picked PRs - we also a Co-Authored-By line (@aweisberg to confirm)

Lastly - this PR changes the type and hence it can possibly break queries and hence we should be adding a Release note at the least which conveys that - 

1. We added a UUID() function
2. For existing accumulo users - the type has changed from VARCHAR to UUID and they need to do an explicit cast now

File: presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloPlugin.java (hunk 1 above)
Comment (ericyuliu) [reply presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloPlugin.java]: Hi @mayankgarg1990 ,Thanks for the suggestions, changes are made accordingly.

**Comment by aweisberg** [comment]
The review and commit guidelines are here and the attribution section covers cherry picking code. https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests

**Comment by aweisberg** [comment]
The release notes have a specific format that is picked up by a script to generate our release notes. The release note guidelines are here https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines

You need to keep it in the triple backticks. They are also sorted by section so you need to specify sections.

A couple of things that make backports easier to review is to link the actual source PR in both the commit message and this PR. It is also sometimes helpful to not merge together multiple commits from Trino.

I see a change on the Trino side marking this UDF as non-deterministic not just by annotation but in a DeterminismEvaluator class. It looks like maybe we don't need it because we are able to get it from the annotation in our version `RowExpressionDeterminismEvaluator`.

The documentation looks somewhat incomplete in that the type isn't documented although the function is. https://github.com/electrum/trino/commit/272a4d664cabaecfce0f9fe4b02b5e6420fe6e3d#diff-9a5c7495c6dee79881154ee22ab20a393207f47c5da547d7b1ba9ee5cae7d25fR240

**Review by mayankgarg1990 — COMMENTED** [review]
On top of what @aweisberg wrote about the release notes - this is what I will recommend the notes to be - 

```
General changes
* Add :func:`uuid` to return a unique identifier

Accumulo Connector Changes
* Replace :func:`uuid` from being connector specific to a standard function. In the process the return type has been changed from ``VARCHAR`` to ``UUID`` and existing use cases might need to perform a cast - ``CAST(UUID() AS VARCHAR)``.
```

**Comment by ericyuliu** [comment]
Hi @aweisberg, some of the changes made on Trino side for ```DeterminismEvaluato``` are skipped in this PR, because just as you mentioned, ```RowExpressionDeterminismEvaluator``` is the new one, and the old hard-coded logic in original trino PR is not needed anymore.

**Review by aweisberg — APPROVED** [review]
OK, LGTM. Thank you!

**Review by aweisberg — CHANGES_REQUESTED** [review]
Can you rebase and see if that fixes the tests? I know at least the time ones were fixed on master.

**Comment by ericyuliu** [comment]
Yep, just rebased and kicked off the workflow

**Comment by ericyuliu** [comment]
@aweisberg Tests are passed!

**Review by aweisberg — APPROVED** [review]

**Review by highker — APPROVED** [review]

**Pull request merged**